axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/SaleOrderServiceSupplychainImpl.java

```diff
@@ -429,4 +430,27 @@ protected void setDefaultInvoicedAndDeliveredPartners(
       saleOrder.setDeliveredPartner(clientPartner);
     }
   }
+
+  public Integer getInvoicingState(SaleOrder saleOrder) {
+
+    BigDecimal amountInvoiced = saleOrder.getAmountInvoiced();
+    BigDecimal exTaxTotal = saleOrder.getExTaxTotal();
+
+    if (amountInvoiced.compareTo(BigDecimal.ZERO) > 0 && amountInvoiced.compareTo(exTaxTotal) < 0) {
+      return StockMoveRepository.STATUS_PARTIALLY_INVOICED;
+    } else if ((amountInvoiced.compareTo(BigDecimal.ZERO) > 0
+            && amountInvoiced.compareTo(exTaxTotal) >= 0)
+        || Beans.get(InvoiceRepository.class)
```

Comment:
chnage condition : wrap this or condition like this
|| ( exTaxTotal ==0 && Beans......count() > 0 )